### PR TITLE
Support indexed bgzipped fasta in variant normalizer

### PR DESCRIPTION
**Pull request by kianfar77**
Signed-off-by: kianfar77 <kiavash.kianfar@databricks.com>

## What changes are proposed in this pull request?
Due to use of ``htsjdk.samtools.reference.IndexedFastaSequenceFile`` to provide the reference genome to variant normalizer, blocked-gzipped-indexed files were not supported. This is now replaced with ``htsjdk.samtools.reference.ReferenceSequenceFileFactory`` which can handle both uncompressed fasta (with .fai) and block-gzipped fasta (with .fai and .gzi). 

Unit tests were added to check normalization works with both uncompressed and bgzipped fastas.

Addresses #176 

## How is this patch tested?
- [x] Unit tests
- [ ] Integration tests
- [ ] Manual tests

(Details)

### CODE REVIEW

**Comment by kianfar77** [comment]
Added a couple of more tests for exceptions at nonexistent path and reference file without index.

File: core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala [hunk 1]
```diff
@@ -199,12 +204,50 @@ class NormalizeVariantsTransformerSuite extends GlowBaseTest with GlowLogging {
...
+    assertThrows[SparkException](
+      testNormalizedvsExpected(
+        vtTestVcfMultiAllelic,
+        vtTestVcfMultiAllelicExpectedNormalized,
+        Option(vtTestReferenceNonexistent)
+      )
+    )
+  }
+
+  gridTest("Reference not indexed ")(
```
Comment (karenfeng) [inline core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala]: Nit: trailing space

File: core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala [hunk 2]
```diff
@@ -199,12 +204,50 @@ class NormalizeVariantsTransformerSuite extends GlowBaseTest with GlowLogging {
...
+      testNormalizedvsExpected(
+        vtTestVcfMultiAllelic,
+        vtTestVcfMultiAllelic,
+        None
+      )
+    )
+  }
+
+  test("Invalid reference path ") {
+    assertThrows[SparkException](
```
Comment (karenfeng) [inline core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala]: For these tests, can we check the exception's error message to make sure the user sees something reasonable?

Comment (kianfar77) [reply core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala]: I have checked. They see proper messages from htsjdk.

Comment (karenfeng) [reply core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala]: Could we catch the error messages with `intercept` to verify in the unit tests?

File: core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala (hunk 1 above)
Comment (kianfar77) [reply core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala]: Fixed.

File: core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala (hunk 2 above)
Comment (kianfar77) [reply core/src/test/scala/io/projectglow/transformers/normalizevariants/NormalizeVariantsTransformerSuite.scala]: Done

**Pull request merged**